sirepo/auth/ldap.py

```diff
@@ -0,0 +1,44 @@
+"""LDAP login
+:copyright: Copyright (c) 2018-2022 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern.pkdebug import pkdlog, pkdp
+from pykern.pkcollections import PKDict
+import ldap
+import sirepo.quest
+
+AUTH_METHOD = "ldap"
+
+#: User can see it
+AUTH_METHOD_VISIBLE = True
+
+
+class API(sirepo.quest.API):
+    @sirepo.quest.Spec("require_cookie_sentinel")
+    def api_authLdapAuthorized(self):
+        """Python-LDAP client authentication
+        Bind to server and authenticate
+        """
+        # Test User Credentials:
+        # username: Archimedes of Syracuse
+        # password: eureka
+
+        LDAP_SERVER = "ldap://10.10.10.10:389"
+        BASE_DN = "dc=example,dc=com"
+        OBJECT_TO_SEARCH = "objectClass=inetOrgPerson"
+        ATTRIBUTES_TO_SEARCH = ["uid"]
+
+        req = self.parse_post(type=False)
+
+        # initialize our connect to ldap-server
+        connect = ldap.initialize(LDAP_SERVER)
```

Comment:
Probably should be an object or cfg.

Reply:
I understand putting the server address in the cfg so it can be changed, @robnagler are you saying to initialize the server in the cfg as well (c = ldap.initialize(LDAP_SERVER))?

Reply:
This is module level initialization for pkconfig. It would look something like:

```py
cfg = pykern.pkconfig.init(
    domain=pykern.pkconfig.RequiredUnlessDev("example.com", str, "LDAP domain"),
)
```